common/finalizer/finalizer.go

```diff
@@ -123,6 +123,10 @@ func (f *Finalizer) Run(
 	f.finalized = true
 	f.mu.Unlock() // unlocking immediately to unblock any calls to Register/Deregister
 
+	// release all references to callbacks at the end to prevent holding on to them for longer than needed
+	// (safe since any calls to Register/Deregister will be aborted)
+	defer func() { f.callbacks = nil }()
```

Comment:
I think there's a potential memory leak here that this addresses:

- say, the finalizer has a few registered callbacks
- each callback is a closure that references the cache item, which itself references the finalizer (see cache.go)
- once a finalizer is executed, it is "sealed" off in that it doesn't allow to add/remove any callbacks anymore
- the issue is that cache items that are evicted from the cache cannot remove themselves from the (now "sealed") finalizer anymore
- therefore, until _all_ cache items that are registered with the finalizer are evicted, the finalizer cannot be garbage collected
- this will likely happen eventually, but by clearing the callbacks after the run, the process is expedited